verifier/src/tcb.rs

```diff
@@ -278,10 +280,79 @@ impl<'a> TryFrom<&'a str> for TcbInfoRaw<'a> {
     }
 }
 
+/// Verifier for ensuring a raw TCB was signed with the provided key
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct TcbInfoRawVerifier {
+    key: VerifyingKey,
+    time: DateTime,
+}
+
+impl TcbInfoRawVerifier {
+    /// Create a new instance.
+    ///
+    /// The `TcbInfoRawVerifier::verify()` will fail if the signature doesn't
+    /// match for the `key` or if the `time` is outside the `issueDate` and
+    /// `nextUpdate` times in the `tcbInfo`.
+    ///
+    /// # Arguments
+    /// - `key` - The public key to verify the `tcbInfo` signature with
+    ///   This should be retrieved from the x509 certificate provided in the
+    ///   TCB request from
+    ///   <https://api.trustedservices.intel.com/sgx/certification/v4/tcb?fmspc={}>
+    /// - `time` - The current system time
+    ///   This is expected to be generated by:
+    ///     ```ignore
+    ///     let time = DateTime::from_system_time(SystemTime::now()).unwrap();
+    ///     ```
+    ///   or equivalent
+    pub fn new(key: VerifyingKey, time: DateTime) -> Self {
+        Self { key, time }
+    }
+}
+
+impl<'a, E: Accessor<TcbInfoRaw<'a>>> Verifier<E> for TcbInfoRawVerifier {
+    type Value = Error;
+    fn verify(&self, evidence: &E) -> VerificationOutput<Self::Value> {
+        let tcb_raw = evidence.get();
+        let result = tcb_raw.verify(&self.key, self.time);
+        let is_success = result.is_ok() as u8;
+
+        // The error will only ever be looked at in failure so the initial
+        // value shouldn't matter.
+        let mut error = Error::SignatureDecodeError;
```

Comment:
Not sure I like this initializing to an error when there may not be one. Technically there is nothing stopping someone from accessing this value when `is_success()` is true.
It could be that we add an unknown or similar, I wouldn't want an `Error::NoError` that feels odd.

Reply:
`Option<Error>`?

Reply:
yup, that's better, Thanks!
I think I avoid `expect()` more than I should to help use and enforce `Option` and `Result` types